Automattic/wp-calypso
# A4A: Implement Hosting page v3 header.

This PR implements the new Hosting Page v3 header.


https://github.com/user-attachments/assets/ca21f75e-2566-41c7-a365-b1ab937247b4

Closes https://github.com/Automattic/automattic-for-agencies-dev/issues/1545

## Proposed Changes

* Implement MigrationOfferV3 banner based on the new design.
* Override certain layout styles to enable scrolling in the main layout container. We also implement an onScroll event listener to connect our expanding and compacting header with the scroll position.
* Introduce the new header featuring key animations when switching to compact mode. In this mode, the title and migration offer are hidden at specific scroll positions.

## Why are these changes being made?

*  This is part of the In-product hosting page v3 project.

## Testing Instructions
 
* Use the A4A live link and go to the `/marketplace/hosting` page.
* Confirm you see the new header and scrolling down works.
* Test that the scrolling works. (Note that there are a few responsiveness issues, but this will be addressed in a separate PR).

## Pre-merge Checklist
